x-pack/test/alerting_api_integration/common/fixtures/plugins/actions_simulators/server/servicenow_simulation.ts

```diff
@@ -7,26 +7,71 @@
 
 import http from 'http';
 
-export const initPlugin = async () => http.createServer(handler);
+/**
+ * This server records the data from a create incident request. It only saves the most recent request. When building tests,
+ * each test should create a new server to avoid stale or overwritten data.
+ */
+export class RecordingServiceNowSimulator {
```

Comment:
This simulator let's me store the incident after a push is called so I can compare it to an expected value.